### PR TITLE
Implement DELEGATECALL (EIP-4)

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: libevm/VM.cpp [hunk 1]
```diff
@@ -66,6 +66,10 @@ void VM::checkRequirements(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp,
 	static const auto c_metrics = metrics();
 	auto& metric = c_metrics[static_cast<size_t>(_inst)];
 
+	// Pre-homstead
```
Comment (LefterisJP) [inline libevm/VM.cpp:69]: Pre-homestead

File: libevm/VM.cpp [hunk 2]
```diff
@@ -153,14 +157,22 @@ void VM::checkRequirements(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp,
...
 			runGas += m_schedule.callNewAccountGas;
-		if (m_stack[m_stack.size() - 3] > 0)
+
+		if (_inst != Instruction::DELEGATECALL && m_stack[m_stack.size() - 3] > 0)
 			runGas += m_schedule.callValueTransferGas;
-		newTempSize = std::max(memNeed(m_stack[m_stack.size() - 6], m_stack[m_stack.size() - 7]), memNeed(m_stack[m_stack.size() - 4], m_stack[m_stack.size() - 5]));
-		break;
 
+		unsigned sizesOffset = _inst == Instruction::DELEGATECALL ? 3 : 4;
+		newTempSize = std::max(memNeed(m_stack[m_stack.size() - sizesOffset - 2], m_stack[m_stack.size() - sizesOffset - 3]), memNeed(m_stack[m_stack.size() - sizesOffset], m_stack[m_stack.size() - sizesOffset - 1]));
```
Comment (LefterisJP) [inline libevm/VM.cpp:173]: Suggestion: Break arguments to `std::max()` each one being in its own line. Also if we are using namespace std you can omit the namespace.

**Comment by gavofyork** [comment]
doh.

**Comment by gavofyork** [comment]
wrong PR.

**Pull request merged**